runtime/src/accounts_db.rs

```diff
@@ -808,16 +810,32 @@ impl AccountsDB {
         Self::hash_account_data(
             slot,
             account.account_meta.lamports,
+            account.account_meta.executable,
+            account.account_meta.rent_epoch,
             account.data,
             &account.meta.pubkey,
         )
     }
 
     pub fn hash_account(slot: Slot, account: &Account, pubkey: &Pubkey) -> Hash {
-        Self::hash_account_data(slot, account.lamports, &account.data, pubkey)
+        Self::hash_account_data(
+            slot,
+            account.lamports,
+            account.executable,
+            account.rent_epoch,
+            &account.data,
+            pubkey,
+        )
     }
 
-    pub fn hash_account_data(slot: Slot, lamports: u64, data: &[u8], pubkey: &Pubkey) -> Hash {
+    pub fn hash_account_data(
```

Comment:
What do you think about passing the entire `account: &Account` into this function rather than the 4 account fields?

Reply:
Thought about the same, too! But there is a reason for the status quo: This function should be accessible from the other code path (no `Account`s there): https://github.com/solana-labs/solana/pull/7415/files#diff-2099c5256db4eb5975c8834af38f6456R810

Of course, we can always normalize to `Account` by `stored_account.clone_account()`, but performance might be affected or rustc is clever enough?

Reply:
Oh ok.  I think the current code is fine then, probably not worth getting fancy yet.  Maybe when we add the 5th or 6th account field later :)